[wip]feat(react-chart): add zoom and pan for chart (#1917)
DevExpress/devextreme-reactive: packages/dx-react-chart/docs/guides/zoom-and-pan.md

# React Chart - Zoom and Pan

Users can zoom and pan (scroll) the React Chart. These actions change the chart's viewport. Mouse and touch gestures are supported.

## Basic Setup

Import the [ZoomAndPan](../reference/zoom-and-pan.md) plugin and add it to the `Chart` component to enable zooming and panning.

Note that the [plugin order](./plugin-overview.md#plugin-order) is important.

### Uncontrolled Mode

In uncontrolled mode, add the `ZoomAndPan` plugin to the `Chart` component.

.embedded-demo({ "path": "zoom-pan/uncontrolled", "showThemeSelector": true })

### Controlled Mode

> In controlled mode, pass the viewport's boundaries to the `ZoomAndPan` plugin's `viewport` property and handle `onViewportChange` event to control zoom and pan externally.

Reviewer: "to control zoom and pan externally" What do you mean by 'externally'?

Reviewer 2: This is a React thing. Components support uncontrolled and controlled modes. In uncontrolled mode, the component's state is manged by the component itself. In controlled mode, the component's state is controlled from outside, hence "controlled" mode.

Reviewer: ```suggestion
In controlled mode, pass the viewport's boundaries to the `ZoomAndPan` plugin's `viewport` property and handle `onViewportChange` event to control zoom and pan.
```

.embedded-demo({ "path": "zoom-pan/controlled", "showThemeSelector": true })

## Enable Zooming or Panning Inidividually

> Use the `interactionWithArguments` and `interactionWithValues` properties to enable zooming or panning individually for corresponding scales.

Reviewer: "zooming or panning individually" Explain.

Reviewer 2: You can enable either zooming or panning or both for individual scales.

Reviewer: You can enable zooming or panning or both for individual scales.
```suggestion
Use the `interactionWithArguments` and `interactionWithValues` properties to enable zooming or panning or both for individual scales.
```

.embedded-demo({ "path": "zoom-pan/datetime", "showThemeSelector": true })

## Zoom by Selecting a Region

> Users can zoom a region by selecting it with the mouse drag gesture while pressing the `zoomRegionKey`. Zooming is applied when the mouse button is released.

Reviewer: ```suggestion
Users can drag the mouse across an area while pressing the `zoomRegionKey` to select the area they want to zoom. Zooming is applied when the mouse button is released.
```

Reviewer: Ok
